src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -501,7 +495,7 @@ private static string CreateProjectActionId()
             Assumes.NotNull(solutionManager);
 
             Dictionary<string, NuGetProject>? projects = (await solutionManager.GetNuGetProjectsAsync())
-                .ToDictionary(project => project.GetMetadata<string>(NuGetProjectMetadataKeys.ProjectId), _ => _);
+                .ToDictionary(project => project.GetMetadata<string>(NuGetProjectMetadataKeys.ProjectId), _ => _, PathUtility.GetStringComparerBasedOnOS());
```

Comment:
This doesn't look right.  Project ID's are GUID's not file system paths.  I think this should be `StringComparer.OrdinalIgnoreCase` like [it was](https://github.com/NuGet/NuGet.Client/pull/3703/files#diff-b1db843d684b7d235cd2b07f4917eafaL556).

That said, I'm surprised you'd need it.  Do you have multiple projects with the same project ID but differing in casing?

Reply:
Sorry, I didn't realize it was just GUID. I'll fix it.

Reply:
Fixed it.